tests/unittests/tests-spiffs/tests-spiffs.c

```diff
@@ -0,0 +1,413 @@
+    .init = _init,
+    .read = _read,
+    .write = _write,
+    .erase = _erase,
+    .power = _power,
+};
+
+static mtd_dev_t dev = {
+    .driver = &driver,
+    .sector_count = SECTOR_COUNT,
+    .pages_per_sector = PAGE_PER_SECTOR,
+    .page_size = PAGE_SIZE,
+};
+
+static mtd_dev_t *_dev = (mtd_dev_t*) &dev;
+#endif /* MTD_0 */
+
+static struct spiffs_desc spiffs_desc = {
+    .lock = MUTEX_INIT,
+};
+
+static vfs_mount_t _test_spiffs_mount = {
+    .fs = &spiffs_file_system,
+    .mount_point = "/test-spiffs",
+    .private_data = &spiffs_desc,
+};
+
+static void test_spiffs_setup(void)
+{
+#if SPIFFS_HAL_CALLBACK_EXTRA == 1
+    spiffs_desc.dev = _dev;
+#endif
+}
+
+static void test_spiffs_teardown(void)
+{
+    vfs_mount(&_test_spiffs_mount);
+    vfs_unlink("/test-spiffs/test.txt");
+    vfs_unlink("/test-spiffs/test0.txt");
+    vfs_unlink("/test-spiffs/test1.txt");
+    vfs_unlink("/test-spiffs/a/test2.txt");
+    vfs_umount(&_test_spiffs_mount);
+}
+
+static void test_spiffs_mount_umount(void)
+{
+    int res;
+    res = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_umount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+}
+
+static void tests_spiffs_open_close(void)
+{
+    int res;
+    res = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_open("/test-spiffs/test.txt", O_CREAT | O_RDWR, 0);
+    TEST_ASSERT(res >= 0);
+
+    res = vfs_close(res);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_umount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+}
+
+static void tests_spiffs_write(void)
+{
+    const char buf[] = "TESTSTRING";
+    char r_buf[2 * sizeof(buf)];
+
+    int res;
+    res = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    int fd = vfs_open("/test-spiffs/test.txt", O_CREAT | O_RDWR, 0);
+    TEST_ASSERT(fd >= 0);
+
+    res = vfs_write(fd, buf, sizeof(buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), res);
+
+    res = vfs_lseek(fd, 0, SEEK_SET);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_read(fd, r_buf, sizeof(r_buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), res);
+    TEST_ASSERT_EQUAL_STRING(&buf[0], &r_buf[0]);
+
+    res = vfs_close(fd);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    fd = vfs_open("/test-spiffs/test.txt", O_RDONLY, 0);
+    TEST_ASSERT(fd >= 0);
+
+    res = vfs_read(fd, r_buf, sizeof(r_buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), res);
+    TEST_ASSERT_EQUAL_STRING(&buf[0], &r_buf[0]);
+
+    res = vfs_close(fd);
+    TEST_ASSERT_EQUAL_INT(0, res);
+
+    res = vfs_umount(&_test_spiffs_mount);
+    TEST_ASSERT_EQUAL_INT(0, res);
+}
+
+static void tests_spiffs_write2(void)
+{
+    char buf[256];
+    /*char r_buf[sizeof(buf) + 2];*/
+
+    for (int i = 0; i < 256; i++) {
+        buf[i] = 'A' + (i % 30);
+    }
+
+    int mp = vfs_mount(&_test_spiffs_mount);
+    TEST_ASSERT(mp >= 0);
+
+    int res;
+    for (int j = 0; j < 5; j++) {
```

Comment:
This needs a comment explaining what we want to test.
I don't really see how the magic numbers in this test ensure that we fill the memory and need to erase sectors..